cpp/src/io/orc/stripe_data.cu

```diff
@@ -1031,32 +1040,42 @@ static __device__ int Decode_Decimals(orc_bytestream_s *bs,
     if (t >= num_vals_read and t < num_vals_to_read) {
       auto const pos = static_cast<int>(vals.i64[t]);
       int128_s v     = decode_varint128(bs, pos);
-      // Since cuDF column stores just one scale, value needs to
-      // be adjusted to col_scale from val_scale. So the difference
-      // of them will be used to add 0s or remove digits.
-      int32_t scale = (t < numvals) ? col_scale - val_scale : 0;
-      if (scale >= 0) {
-        scale       = min(scale, 27);
-        vals.i64[t] = ((int64_t)v.lo * kPow5i[scale]) << scale;
-      } else  // if (scale < 0)
-      {
-        bool is_negative = (v.hi < 0);
-        uint64_t hi = v.hi, lo = v.lo;
-        scale = min(-scale, 27);
-        if (is_negative) {
-          hi = (~hi) + (lo == 0);
-          lo = (~lo) + 1;
-        }
-        lo = (lo >> (uint32_t)scale) | ((uint64_t)hi << (64 - scale));
-        hi >>= (int32_t)scale;
-        if (hi != 0) {
-          // Use intermediate float
-          lo = __double2ull_rn(Int128ToDouble_rn(lo, hi) / __ll2double_rn(kPow5i[scale]));
-          hi = 0;
-        } else {
-          lo /= kPow5i[scale];
+
+      if (col_scale & orc_decimal2float64_scale) {
```

Comment:
Is this the same as the old code? I'd rather not dig into this if it's already field tested :grimacing:

Reply:
Yes, I literally copied and pasted it from the previous commit.